## Sorting Rows 

"The NBA is the highest paying professional sports league in the world," reported CNN in March 2016. The table nba_salaries contains the salaries of all National Basketball Association players in 2015-2016.

Each row represents one player. The columns are:

In [4]:
import pandas as pd
import numpy as np

In [13]:
nba_salaries=pd.read_csv(r'E:\mtech AI\Data Science\Completed Assignment\Chapter 6\nbasalaries.csv')
nba_salaries

PLAYER POSITION                TEAM  '15-'16 SALARY
0       Paul Millsap       PF       Atlanta Hawks       18.671659
1         Al Horford        C       Atlanta Hawks       12.000000
2     Tiago Splitter        C       Atlanta Hawks        9.756250
3        Jeff Teague       PG       Atlanta Hawks        8.000000
4        Kyle Korver       SG       Atlanta Hawks        5.746479
..               ...      ...                 ...             ...
412        Gary Neal       PG  Washington Wizards        2.139000
413     DeJuan Blair        C  Washington Wizards        2.000000
414  Kelly Oubre Jr.       SF  Washington Wizards        1.920240
415   Garrett Temple       SG  Washington Wizards        1.100602
416     Jarell Eddie       SG  Washington Wizards        0.561716

[417 rows x 4 columns]

The table contains 417 rows, one for each player. Only 10 of the rows are displayed. The show method allows us to specify the number of rows, with the default (no specification) being all the rows of the table.

In [15]:
nba_salaries.iloc[:3]

PLAYER POSITION           TEAM  '15-'16 SALARY
0    Paul Millsap       PF  Atlanta Hawks       18.671659
1      Al Horford        C  Atlanta Hawks       12.000000
2  Tiago Splitter        C  Atlanta Hawks        9.756250

In [16]:
print(nba_salaries.iloc[[3]])

        PLAYER POSITION           TEAM  '15-'16 SALARY
3  Jeff Teague       PG  Atlanta Hawks             8.0


In [18]:
print(nba_salaries.iloc[:],[1,4])

              PLAYER POSITION                TEAM  '15-'16 SALARY
0       Paul Millsap       PF       Atlanta Hawks       18.671659
1         Al Horford        C       Atlanta Hawks       12.000000
2     Tiago Splitter        C       Atlanta Hawks        9.756250
3        Jeff Teague       PG       Atlanta Hawks        8.000000
4        Kyle Korver       SG       Atlanta Hawks        5.746479
..               ...      ...                 ...             ...
412        Gary Neal       PG  Washington Wizards        2.139000
413     DeJuan Blair        C  Washington Wizards        2.000000
414  Kelly Oubre Jr.       SF  Washington Wizards        1.920240
415   Garrett Temple       SG  Washington Wizards        1.100602
416     Jarell Eddie       SG  Washington Wizards        0.561716

[417 rows x 4 columns] [1, 4]


Glance through about 20 rows or so, and you will see that the rows are in alphabetical order by team name. It's also possible to list the same rows in alphabetical order by player name using the sort method. The argument to sort is a column label or index.

In [23]:
nba_salaries.sort_values(by=['PLAYER'], inplace=True)


In [25]:
nba_salaries

PLAYER POSITION                    TEAM  '15-'16 SALARY
68      Aaron Brooks       PG           Chicago Bulls        2.250000
291     Aaron Gordon       PF           Orlando Magic        4.171680
59    Aaron Harrison       SG       Charlotte Hornets        0.525093
235    Adreian Payne       PF  Minnesota Timberwolves        1.938840
1         Al Horford        C           Atlanta Hawks       12.000000
..               ...      ...                     ...             ...
95   Wilson Chandler       SF          Denver Nuggets       10.449438
233      Zach LaVine       PG  Minnesota Timberwolves        2.148360
181    Zach Randolph       PF       Memphis Grizzlies        9.638555
86     Zaza Pachulia        C        Dallas Mavericks        5.200000
26      Zoran Dragic       SG          Boston Celtics        1.706225

[417 rows x 4 columns]

To examine the players' salaries, it would be much more helpful if the data were ordered by salary.

To do this, we will first simplify the label of the column of salaries (just for convenience), and then sort by the new label SALARY.

In [35]:
nba_salaries.rename(columns = {"'15-'16 SALARY":'SALARY'}, inplace = True)
nba_salaries

PLAYER POSITION                    TEAM     SALARY
68      Aaron Brooks       PG           Chicago Bulls   2.250000
291     Aaron Gordon       PF           Orlando Magic   4.171680
59    Aaron Harrison       SG       Charlotte Hornets   0.525093
235    Adreian Payne       PF  Minnesota Timberwolves   1.938840
1         Al Horford        C           Atlanta Hawks  12.000000
..               ...      ...                     ...        ...
95   Wilson Chandler       SF          Denver Nuggets  10.449438
233      Zach LaVine       PG  Minnesota Timberwolves   2.148360
181    Zach Randolph       PF       Memphis Grizzlies   9.638555
86     Zaza Pachulia        C        Dallas Mavericks   5.200000
26      Zoran Dragic       SG          Boston Celtics   1.706225

[417 rows x 4 columns]

This arranges all the rows of the table in increasing order of salary, with the lowest salary appearing first. The output is a new table with the same columns as the original but with the rows rearranged.

In [37]:
n=nba_salaries.sort_values(by=['SALARY'], ascending=True)
n

PLAYER POSITION                 TEAM     SALARY
267  Thanasis Antetokounmpo       SF      New York Knicks   0.030888
327          Cory Jefferson       PF         Phoenix Suns   0.049709
326            Jordan McRae       SG         Phoenix Suns   0.049709
324         Orlando Johnson       SG         Phoenix Suns   0.055722
325            Phil Pressey       PG         Phoenix Suns   0.055722
..                      ...      ...                  ...        ...
131           Dwight Howard        C      Houston Rockets  22.359364
255         Carmelo Anthony       SF      New York Knicks  22.875000
72             LeBron James       SF  Cleveland Cavaliers  22.970500
29              Joe Johnson       SF        Brooklyn Nets  24.894863
169             Kobe Bryant       SF   Los Angeles Lakers  25.000000

[417 rows x 4 columns]

To order the rows of the table in decreasing order of salary, we must use sort with the option descending=True.

In [38]:
n=nba_salaries.sort_values(by=['SALARY'], ascending=False)
n

PLAYER POSITION                 TEAM     SALARY
169             Kobe Bryant       SF   Los Angeles Lakers  25.000000
29              Joe Johnson       SF        Brooklyn Nets  24.894863
72             LeBron James       SF  Cleveland Cavaliers  22.970500
255         Carmelo Anthony       SF      New York Knicks  22.875000
131           Dwight Howard        C      Houston Rockets  22.359364
..                      ...      ...                  ...        ...
324         Orlando Johnson       SG         Phoenix Suns   0.055722
325            Phil Pressey       PG         Phoenix Suns   0.055722
327          Cory Jefferson       PF         Phoenix Suns   0.049709
326            Jordan McRae       SG         Phoenix Suns   0.049709
267  Thanasis Antetokounmpo       SF      New York Knicks   0.030888

[417 rows x 4 columns]

Named Arguments 
The descending=True portion of this call expression is called a named argument. When a function or method is called, each argument has both a position and a name. Both are evident from the help text of a function or method.

In [39]:
help(nba_salaries.sort_values)

Help on method sort_values in module pandas.core.frame:

sort_values(by, axis=0, ascending=True, inplace=False, kind='quicksort', na_position='last', ignore_index=False) method of pandas.core.frame.DataFrame instance
    Sort by the values along either axis.
    
    Parameters
    ----------
            by : str or list of str
                Name or list of names to sort by.
    
                - if `axis` is 0 or `'index'` then `by` may contain index
                  levels and/or column labels.
                - if `axis` is 1 or `'columns'` then `by` may contain column
                  levels and/or index labels.
    
                .. versionchanged:: 0.23.0
    
                   Allow specifying index or column level names.
    axis : {0 or 'index', 1 or 'columns'}, default 0
         Axis to be sorted.
    ascending : bool or list of bool, default True
         Sort ascending vs. descending. Specify list for multiple sort
         orders.  If this is a list of bools, must